core/src/test/scala/org/apache/spark/rdd/RDDSuite.scala

```diff
@@ -833,6 +833,30 @@ class RDDSuite extends SparkFunSuite with SharedSparkContext {
     }
   }
 
+  test("zipWithIndex with partition size exceeding MaxInt") {
+    val result = sc.parallelize(Seq(1), 1).mapPartitions(
```

Comment:
yeah, here I run the test, because the loop of CPU is very fast, running this testcase on my machine is about 3s. Does it still need optimized ?